platform/ios/docs/guides/Data-Driven Styling.md

```diff
@@ -0,0 +1,127 @@
+
+![available bikes](img/data-driven-styling/citibikes.png)
+
+
+_Insert cool overview of things you can do!_
+
+### How to use Data-Driven Styling
+This guide uses earthquake data from the [U.S. Geological Survey](https://earthquake.usgs.gov/earthquakes/feed/v1.0/geojson.php) to illustrate to style a map based on attributes. For more information about how to work with GeoJSON data in our iOS SDK, please see our [working with GeoJSON data](https://www.mapbox.com/ios-sdk/api/3.5.0-beta.1/working-with-geojson-data.html) guide.
+
+`MGLStyleFunction`
+
+There are three subclasses of `MGLStyleFunction`:
+
+* `MGLCameraStyleFunction` - For a style value that changes with zoom level. For example, you can make the radius of a circle to increase based on zoom level. In iOS SDK v3.4, a `MGLStyleFunction` referred to what is now a `MGLCameraStyleFunction`.
+* `MGLSourceStyleFunction` - For a style value that changes with the attributes of a feature. You can adjust the radius of a circle based on the magnitude of an earthquake, for example.
+* `MGLCompositeStyleFunction` - For a style value that changes with both zoom level and attribute values. For example, you can add a circle layer where each circle has a radius based on both zoom level and the magnitude of an earthquake.
+
+####Stops
+
+Stops are key-value pairs that that determine a style value. With a `MGLCameraSourceFunction` stop, you can use a dictionary with a zoom level for a key and a `MGLStyleValue` for the value. For example, you can use a stops dictionary with zoom levels 0, 10, and 20 as keys, and yellow, orange, and red as the values. A `MGLSourceStyleFunction` uses the relevant attribute value as the key.
+
+####`MGLInterpolationMode`
+
+The effect a key has on the style value is determined by the interpolation mode. There are four interpolation modes that can be used with a source style function: exponential, interval, categorical, and identity. You can also use exponential and interval interpolation modes with a camera style function.
+
+#####Exponential:
+
+`MGLInterpolationModelExponential` creates a linear effect based on the values. The key value is the base for interpolation, and the style value is based on where an attribute value falls between two keys.
+
+The stops dictionary below, for example, shows colors that continuously shift from yellow -> orange -> red -> blue -> white based on the attribute value.
+
+```
+  if let url = URL(string: "https://earthquake.usgs.gov/earthquakes/feed/v1.0/summary/all_week.geojson") {
+            let symbolSource = MGLSource(identifier: "source")
+            let symbolLayer = MGLSymbolStyleLayer(identifier: "place-city-sm", source: symbolSource)
+
+            let source = MGLShapeSource(identifier: "earthquakes", url: url, options: nil)
+            style.addSource(source)
+
+            let stops = [0 : MGLStyleValue(rawValue: UIColor.yellow),
```

Comment:
Point taken about MGLColor being pedagogically problematic. In the Mapbox API Documentation examples in Objective-C, I took to using conditional compilation macros, which are better documented but even more poorly understood by less experienced developers.

What about bringing this guide under the same codegen mechanism that the “Information for Style Authors” guide uses?

1. Move this guide to platform/darwin/docs/guides/Data-Driven Styling.md.ejs.
1. Define a [`cocoaPrefix`](https://github.com/mapbox/mapbox-gl-native/blob/a0a69ad6a7d4428961d99564b9b09b23a0acfcf6/platform/darwin/docs/guides/For%20Style%20Authors.md.ejs#L5) variable at the top.
1. Put a `<% cocoaPrefix %>` variable anywhere we’re currently saying `UI`.
1. Extend generate-style-code.js to also [write out](https://github.com/mapbox/mapbox-gl-native/blob/a0a69ad6a7d4428961d99564b9b09b23a0acfcf6/platform/darwin/scripts/generate-style-code.js#L607-L616) iOS- and macOS-specific versions of the guide.
1. Run `make style-code`.

With this approach, the style-related guides would be more consistent with each other even at the code level. Additionally, it’ll give us the ability to put some of these (rather substantial) code examples in unit tests and insert them into this document automatically, ensuring that the examples stay current. We already use that approach for examples embedded in API documentation.